src/typescript.js

```diff
@@ -0,0 +1,110 @@
+// @ts-check
+
+// @ts-expect-error -- False positive
+import importPlugin from 'eslint-plugin-import';
+// eslint-disable-next-line import/no-unresolved -- False positive
+import typescriptEslint from 'typescript-eslint';
+import { ERROR, OFF, typescriptFiles } from './config.js';
+import javascript from './javascript.js';
+
+/**
+ * @type {Array<import('eslint').Linter.Config>}
+ */
+export default [
+  ...javascript,
+
+  typescriptEslint.configs.eslintRecommended,
+  ...typescriptEslint.configs.recommended,
```

Comment:
After https://github.com/tools-aoeur/component-library/pull/1954 we should add
```typescript
...typescriptEslint.configs.stylisticTypeChecked
```
here.